sdk/src/transaction/versioned/sanitized.rs

```diff
@@ -11,27 +14,36 @@ pub struct SanitizedVersionedTransaction {
     pub(crate) signatures: Vec<Signature>,
     /// Message to sign.
     pub(crate) message: SanitizedVersionedMessage,
-}
-
-impl TryFrom<VersionedTransaction> for SanitizedVersionedTransaction {
-    type Error = SanitizeError;
-    fn try_from(tx: VersionedTransaction) -> Result<Self, Self::Error> {
-        Self::try_new(tx)
-    }
+    /// Meta data
+    pub(crate) transaction_meta: TransactionMeta,
 }
 
 impl SanitizedVersionedTransaction {
-    pub fn try_new(tx: VersionedTransaction) -> Result<Self, SanitizeError> {
+    pub fn try_new(
+        tx: VersionedTransaction,
+        feature_set: &FeatureSet,
+    ) -> Result<Self, SanitizeError> {
         tx.sanitize_signatures()?;
+        let sanitized_versioned_message = SanitizedVersionedMessage::try_from(tx.message)?;
+        let transaction_meta = process_compute_budget_instruction(
+            sanitized_versioned_message.program_instructions_iter(),
+            feature_set,
+            None, //cluster type
+        )?;
```

Comment:
process compute_budget instructions once when sanitizing VersionedTransaction, transaction_meta will be moved to SanitizedTransaction.